File: app/src/main/java/com/univreview/model/Professor.java
package com.univreview.model;

import com.google.gson.annotations.Expose;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by DavidHa on 2017. 1. 13..
 */
public class Professor extends AbstractDataProvider implements Serializable{
    @Expose
    public Long id;
    @Expose
    public String name;

    @Expose
    public List<SubjectDetail> subjectDetail = new ArrayList<>();

    @Override
    public long getId() {
        return id;
    }

    @Override
    public String getName() {
        return name;
    }
}

File: app/src/main/java/com/univreview/model/Professor.kt
package com.univreview.model

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by DavidHa on 2017. 8. 20..
 */
data class Professor(override var id: Long, override var name: String) : AbstractDataProvider(), Parcelable {
    constructor(source: Parcel) : this(
            source.readLong(),
            source.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeLong(id)
        writeString(name)
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Professor> = object : Parcelable.Creator<Professor> {
            override fun createFromParcel(source: Parcel): Professor = Professor(source)
            override fun newArray(size: Int): Array<Professor?> = arrayOfNulls(size)
        }
    }
}